contracts/QuotaControlled.sol

```diff
@@ -0,0 +1,108 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/IQuotaControlled.sol";
+
+contract QuotaControlled is IQuotaControlled {
+    struct Checkpoint {
+        uint32 fromTimestamp;
+        uint224 value;
+    }
+    struct Quota {
+        uint32 period;
+        uint224 amount;
+    }
+
+    mapping(address => Checkpoint[])
+        public
+        override accountToAccumulatedQuotaUsageCheckpoints;
+    mapping(address => Quota) public override accountToQuota;
+
+    function getQuotaUsage(address account)
+        public
+        view
+        override
+        returns (uint224)
+    {
+        Checkpoint[]
+            storage accumulatedQuotaUsageCheckpoints = accountToAccumulatedQuotaUsageCheckpoints[
+                account
+            ];
+        uint224 accumulatedQuotaUsage = 0;
+        if (accumulatedQuotaUsageCheckpoints.length > 0) {
+            accumulatedQuotaUsage = accumulatedQuotaUsageCheckpoints[
+                accumulatedQuotaUsageCheckpoints.length - 1
+            ].value;
+        }
+        uint224 accumulatedQuotaUsageThen = getValueAt(
+            accumulatedQuotaUsageCheckpoints,
+            uint32(block.timestamp) - accountToQuota[account].period
+        );
+        return accumulatedQuotaUsage - accumulatedQuotaUsageThen;
+    }
+
+    function _setQuota(
```

Comment:
According to my style, private/internal functions that start with _ are expected to be exposed and vice versa, so these are not random

Reply:
Meaning that the contract inheriting this one will probably have a public/external function that would act as proxy to these functions?

Reply:
Yeah